apps/container/src/components/TeacherReview.vue

```diff
@@ -0,0 +1,165 @@
+            @update:selectedSubject="selectedSubject = $event"
+          />
+        </v-col>
+      </v-row>
+      <div v-else class="d-flex align-center flex-column">
+        <img
+          src="@/assets/comment_not_found.gif"
+          style="width: 100%; max-width: 275px"
+          class="mb-5"
+        />
+        <p>Nenhum dado encontrado 😕</p>
+        <p>
+          Você já fez matéria com esse professor? Se sim, atualize seu histórico
+        </p>
+        <v-btn href="/history" color="primary" class="text-body-1 mt-5">
+          Atualizar
+        </v-btn>
+      </div>
+    </v-container>
+  </PaperCard>
+</template>
+
+<script lang="ts" setup>
+import { useQuery } from '@tanstack/vue-query';
+import { Reviews } from 'services';
+import { transformConceptDataToObject } from 'utils';
+import { computed, ref } from 'vue';
+import { useDisplay } from 'vuetify';
+import FeedbackAlert from './FeedbackAlert.vue';
+import ConceptsPieChart from './ConceptsPieChart.vue';
+
+import CenteredLoading from '@/components/CenteredLoading.vue';
+import PaperCard from '@/components/PaperCard.vue';
+import CommentsList from '@/components/CommentsList.vue';
+
+const props = defineProps({
+  teacherId: { type: String, required: true },
+});
+
+const teacherId = computed(() => props.teacherId);
+
+const { xs } = useDisplay();
+
+const selectedSubject = ref<string>('Todas as matérias');
+
+const {
+  data: teacherData,
+  isFetching: isFetchingTeacher,
+  isError: isFetchingTeacherError,
+} = useQuery({
+  refetchOnWindowFocus: false,
+  queryKey: ['teacher', teacherId],
+  queryFn: () => Reviews.getTeacher(teacherId.value),
+  enabled: !!teacherId.value,
+});
+
+const chips = computed(() => {
+  if (!teacherData.value?.data) {
+    return [];
+  }
+  return [
+    {
+      value: teacherData.value.data.specific.filter(
+        (subject) => subject._id !== null,
+      ).length,
+      text:
+        teacherData.value.data.specific.filter(
+          (subject) => subject._id !== null,
+        ).length == 1
+          ? 'disciplina'
+          : 'disciplinas',
+      icon: 'mdi-human-male-board',
+    },
+    {
+      value:
+        selectedSubject.value === 'Todas as matérias'
+          ? teacherData.value.data.general.count
+          : teacherData.value.data.specific
+              .filter((subject) => subject._id !== null)
+              .find((subject) => subject._id.name === selectedSubject.value)
+              ?.count,
+      text:
+        selectedSubject.value === 'Todas as matérias'
+          ? teacherData.value.data.general.count == 1
+            ? 'conceito'
+            : 'conceitos'
+          : teacherData.value.data.specific
+              .filter((subject) => subject._id !== null)
+              .find((subject) => subject._id.name === selectedSubject.value)
+              ?.count == 1
+          ? 'conceito'
+          : 'conceitos',
+      icon: 'mdi-format-annotation-plus',
+    },
+  ];
+});
```

Comment:
Evite essa quantidade de ternarios aninhados, fiz algumas refatorações, confere se funciona por favor
```suggestion
const chips = computed(() => {
  if (!teacherData.value?.data) {
    return [];
  }
  const { general, specific } = teacherData.value.data;

  const specificValid = specific.filter((subject) => subject._id);

  const specificValidSelected = specificValid.find(
    (subject) => subject._id.name === selectedSubject.value,
  );

  const toPlural = (value?: number) => (value == 1 ? '' : 's');

  const isAllSubjects = selectedSubject.value === 'Todas as matérias';

  return [
    {
      value: specificValid.length,
      text: `disciplina${toPlural(specificValid.length)}`,
      icon: 'mdi-human-male-board',
    },
    {
      value: isAllSubjects ? general.count : specificValidSelected?.count,
      text: isAllSubjects
        ? `conceito${toPlural(general.count)}`
        : `conceito${toPlural(specificValidSelected?.count)}`,
      icon: 'mdi-format-annotation-plus',
    },
  ];
});
```

Reply:
a função to plural poderia receber um parâmetro a mais, tipo suffix, que o padrão é `s` e ir pro `utils` também

Reply:
Isso pode ficar pro futuro